ext/ffi/jit_trampoline.rs

```diff
@@ -58,11 +58,24 @@ fn native_to_c(ty: &NativeType) -> &'static str {
 
 pub(crate) fn codegen(sym: &crate::Symbol) -> String {
   let mut c = String::from(include_str!("prelude.h"));
-  let ret = native_to_c(&sym.result_type);
+  let needs_unwrap = matches!(
+    &sym.result_type,
+    NativeType::Function
+      | NativeType::Pointer
+      | NativeType::I64
+      | NativeType::ISize
+      | NativeType::U64
+      | NativeType::USize
+  );
+  let ret = if needs_unwrap {
+    "void"
+  } else {
+    native_to_c(&sym.result_type)
+  };
 
   // extern <return_type> func(
   c += "\nextern ";
-  c += ret;
+  c += native_to_c(&sym.result_type);
```

Comment:
Rearrange this to unconditionally eval `native_to_c(&sym.result_type)` early, then just use that value in the `else` branch and here?